Fix autograd.functional tests to not fail with logging tensor

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #76057

This PR:
- Set `requires_grad` on the outer tensor. Previously it was set on the inner tensor.
- Passes `is_grads_batched` to `__torch_function__` (not actually necessary to make tests pass if `__torch_function__` is disabled)
- Disable `__torch_function__` for logging tensor.

soulitzer committed:
Fix autograd.functional tests to not fail with logging tensor

[ghstack-poisoned]

albanD commented on torch/testing/_internal/logging_tensor.py:
```diff
@@ -36,6 +36,8 @@ class LoggingTensor(torch.Tensor):
 
     __slots__ = ['elem']
 
+    __torch_function__ = torch._C._disabled_torch_function_impl
```
This should not be needed. There was a change by Ed to make this the default.

albanD reviewed (commented):
Update to the test sounds good.
The disabling of torch_function should not be needed though.

EDIT: actually that fine as this didn't land yet: https://github.com/pytorch/pytorch/pull/73942

soulitzer committed:
Update on "Fix autograd.functional tests to not fail with logging tensor"


This PR:
- Set `requires_grad` on the outer tensor. Previously it was set on the inner tensor.
- Passes `is_grads_batched` to `__torch_function__` (not actually necessary to make tests pass if `__torch_function__` is disabled)
- Disable `__torch_function__` for logging tensor.

[ghstack-poisoned]

soulitzer committed:
Update on "Fix autograd.functional tests to not fail with logging tensor"


This PR:
- Set `requires_grad` on the outer tensor. Previously it was set on the inner tensor.
- Passes `is_grads_batched` to `__torch_function__` (not actually necessary to make tests pass if `__torch_function__` is disabled)
- Disable `__torch_function__` for logging tensor.

[ghstack-poisoned]

soulitzer committed:
Update on "Fix autograd.functional tests to not fail with logging tensor"


This PR:
- Set `requires_grad` on the outer tensor. Previously it was set on the inner tensor.
- Passes `is_grads_batched` to `__torch_function__` (not actually necessary to make tests pass if `__torch_function__` is disabled)
- Disable `__torch_function__` for logging tensor.

[ghstack-poisoned]

soulitzer commented on test/test_functionalization.py:
```diff
@@ -461,14 +461,11 @@ def f(x, y):
         # with a functional + non-functional tensor, and wrapped the output appropriately.
         self.assertExpectedInline('\n'.join(logs), """\
 $2 = torch._ops.aten.add.Tensor($0, $1)
-$3 = torch._ops.aten.alias_copy.default($2)
```
@bdhirsh I'm guessing the reason this went away is because when `__torch_function__` wraps the output of add, it calls `as_subclass` which calls into `alias`, and now that we disable `__torch_function__` for logging tensor it no longer does that.

bdhirsh replied:
That sounds right, and the new trace looks better!

soulitzer commented:
@pytorchbot merge this

soulitzer committed:
Update on "Fix autograd.functional tests to not fail with logging tensor"


This PR:
- Set `requires_grad` on the outer tensor. Previously it was set on the inner tensor.
- Passes `is_grads_batched` to `__torch_function__` (not actually necessary to make tests pass if `__torch_function__` is disabled)
- Disable `__torch_function__` for logging tensor.

[ghstack-poisoned]

soulitzer commented:
@pytorchbot merge this
